nemo_skills/pipeline/utils/declarative.py

```
@@ -492,6 +616,321 @@ def run(self, dry_run: bool = False, log_dir: Optional[str] = None, _reuse_exp=N
+                run_after_list = self.run_after if isinstance(self.run_after, list) else [self.run_after]
+                job_dependencies = run_after_list
+
+            for dep in job_dependencies:
+                if isinstance(dep, str):
+                    # String dependency = external experiment name - not supported on K8s
+                    LOG.warning(f"External dependency '{dep}' not supported on Kubernetes, skipping")
+                elif isinstance(dep, dict):
+                    # Dict dependency = internal job reference (same as _run_nemo_run)
+                    dep_name = dep.get("name")
+                    if not dep_name:
+                        raise ValueError(f"Job dependency must have a 'name' field: {dep}")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)